File: task-03/C++/Sub-Task4.cpp
#include <fstream>
#include <string>

int main() {
    std::ifstream input("input.txt");
    std::ofstream output("output.txt");
    int n;
    input >> n;

    for (int i = 0; i < n; i++) {
        output << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << "\n";
    }
    for (int i = n - 2; i >= 0; i--) {
        output << std::string(n - i - 1, ' ') << std::string(2 * i + 1, '*') << "\n";
    }

    input.close();
    output.close();
    return 0;
}

File: task-03/Rust/Sub-Task4.rs
use std::fs;

fn main() {
    let n: i32 = fs::read_to_string("input.txt")
        .expect("Failed to read file")
        .trim()
        .parse()
        .expect("Invalid number");

    let mut output = String::new();

    for i in 0..n {
        output.push_str(&format!("{}{}\n", " ".repeat((n-i-1) as usize), "*".repeat((2*i+1) as usize)));
    }

    for i in (0..n-1).rev() {
        output.push_str(&format!("{}{}\n", " ".repeat((n-i-1) as usize), "*".repeat((2*i+1) as usize)));
    }

    fs::write("output.txt", output).expect("Failed to write file");
}
